Больше подробностей (часть 1)

Ivan-Pelipenko opened the pull request:
- создание и отрисовка карты

---
:mortar_board: [Больше подробностей (часть 1)](https://up.htmlacademy.ru/react/12/user/2176283/tasks/8)

Ivan-Pelipenko committed:
Больше подробностей (часть 1)
- создание и отрисовка карты

Ivan-Pelipenko committed:
Отрисовка всех предложений

Ivan-Pelipenko committed:
- Отображение выбранного предложения на карте

Ivan-Pelipenko committed:
- исправления ошибок

saimon395 commented on project/src/types/location.ts:
```diff
@@ -0,0 +1,7 @@
+export type Location = {
+  title: string;
+lat: number;
+lng: number;
+}
```
форматирование поправь

saimon395 commented on project/src/types/city.ts:
```diff
@@ -0,0 +1,6 @@
+export type City = {
+  title: string;
+lat: number;
+lng: number;
+zoom: number;
```
здесь тоже

saimon395 commented on project/src/pages/welcome-page/welcome-page.tsx:
```diff
@@ -116,11 +107,15 @@ function WelcomePage({offerCount, offers}: WelcomePageProps): JSX.Element {
               </form>
 
-              <OffersList offers={offers} offerCount={offerCount}/>
+              <OffersList offers={offers} offerCount={offerCount} onListOfferHover={onListOfferHover}/>
 
```
есть критерий про именование обработчиков, проп пусть и дальше называется onListOfferHover, а вот сам обработчик назови через handle
```suggestion
              <OffersList offers={offers} offerCount={offerCount} onListOfferHover={handleListOfferHover}/>

```

saimon395 commented on project/src/pages/welcome-page/welcome-page.tsx:
```diff
@@ -1,13 +1,33 @@
+    const currentOffer = offers.find((offer) =>
+      offer.id === listOfferId,
+    );
+    if(currentOffer) {
+      return setSelectedOffer(currentOffer);
+    }
+    else {
+      return setSelectedOffer('');
+    }
```
если у тебя функция возвращает void, судя по типизации, то зачем здесь вообще return ? тоже самое с типизацией параметра, ты либо оставь общую типизацию после названия функции, либо типизируй listOfferId в круглых скобках

saimon395 commented on project/src/pages/offers-list/offers-list.tsx:
```diff
@@ -1,17 +1,18 @@
   const [ activeCard , setActiveCard ] = useState <number | null>(null);
-  // eslint-disable-next-line no-console
-  console.log(activeCard);
+
+  useEffect(() => {onListOfferHover(activeCard);},);
```
а точно ли здесь именно useEffect нужен ?

saimon395 replied:
да ещё и без массива зависимостей

saimon395 replied:
и если у тебя есть стэйт в WelcomePage, отсюда его можно убрать, зачем тебе хранить активную карточку и тут и там ?

saimon395 commented on project/src/components/map/map.tsx:
```diff
@@ -0,0 +1,58 @@
+  const map = useMap(mapRef, city);
+
+  const defaultCustomIcon = leaflet.icon({
+    iconUrl: 'https://assets.htmlacademy.ru/content/intensive/javascript-1/demo/interactive-map/pin.svg',
+    iconSize: [40, 40],
+    iconAnchor: [20, 40],
+  });
+
+  const currentCustomIcon = leaflet.icon({
+    iconUrl: 'https://assets.htmlacademy.ru/content/intensive/javascript-1/demo/interactive-map/main-pin.svg',
```
метки используй те что вместе с проектом идут , в папке img

Ivan-Pelipenko committed:
- правка 'lat' и 'lng'
- правка именования обработчика
Задание 5.2
- создание новых компонентов «Список отзывов» и «Отзыв»
- обновление компонента 'offer-used-logged-page'(Room)

Ivan-Pelipenko committed:
- правка функции 'handleListOfferHover'
- убрал состояние из 'offers-list'

Ivan-Pelipenko committed:
- использование меток проекта

saimon395 commented on project/src/hooks/use-map/use-Map.ts:
```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState, useRef, MutableRefObject } from 'react';
+import leaflet from 'leaflet';
+import { Map } from 'leaflet';
```
переименуй файл, что бы в названии не было больших букв

saimon395 commented on project/src/components/map/map.tsx:
```diff
@@ -0,0 +1,58 @@
+  const map = useMap(mapRef, city);
+
+  const defaultCustomIcon = leaflet.icon({
+    iconUrl: './img/pin.svg',
+    iconSize: [40, 40],
+    iconAnchor: [20, 40],
+  });
+
+  const currentCustomIcon = leaflet.icon({
+    iconUrl: './img/pin-active.svg',
+    iconSize: [40, 40],
+    iconAnchor: [20, 40],
+  });
```
размеры иконок поправь, а то они теперь по высоте сплюснутые
